### PR TITLE
Implementation of ForumService, Repository and connection to the frontend.

### PR DESCRIPTION
**Pull request by MarcoSteinke**

### CODE REVIEW

File: src/main/java/mops/foren/domain/repositoryabstraction/IPostRepository.java [hunk 1]
```diff
@@ -1,4 +1,16 @@
 package mops.foren.domain.repositoryabstraction;
 
+import mops.foren.domain.model.*;
+import mops.foren.domain.model.Thread;
+
+import java.util.List;
+
 public interface IPostRepository {
+    List<Post> getPostsFromThread(Thread thread);
+
```
Comment (niruc100) [inline src/main/java/mops/foren/domain/repositoryabstraction/IPostRepository.java]: redundant

File: src/main/java/mops/foren/domain/repositoryabstraction/IThreadRepository.java [hunk 2]
```diff
@@ -1,4 +1,16 @@
...
+import mops.foren.domain.model.Thread;
+import mops.foren.domain.model.ThreadId;
+import mops.foren.domain.model.Topic;
+import mops.foren.domain.model.TopicId;
+
+import java.util.List;
+
 public interface IThreadRepository {
+    List<Thread> getThreadsFromTopic(Topic topic);
+
```
Comment (niruc100) [inline src/main/java/mops/foren/domain/repositoryabstraction/IThreadRepository.java]: redundant

File: src/main/java/mops/foren/infrastructure/persistence/dtos/UserDTO.java [hunk 3]
```diff
@@ -23,6 +23,14 @@
 
     private String email;
 
+
+    @ManyToMany(fetch = FetchType.EAGER, cascade = CascadeType.ALL)
+    @JoinTable(
```
Comment (niruc100) [inline src/main/java/mops/foren/infrastructure/persistence/dtos/UserDTO.java]: maybe change CascadeType.ALL

File: src/main/java/mops/foren/infrastructure/persistence/mapper/UserMapper.java [hunk 4]
```diff
@@ -38,4 +23,26 @@ public static UserDTO mapUserToUserDto(User user) {
...
+                .email(userDTO.getEmail())
+                .forums(forenIds)
+                .build();
+    }
+
+    private static List<ForumId> getForumIdFromUserDto(UserDTO userDTO) {
+        return userDTO.getForums().stream()
+                .map(ForumMapper::mapForumDtoToForum)
+                .map(forum -> new ForumId(forum.getId().getId()))
+                .collect(Collectors.toList());
```
Comment (niruc100) [inline src/main/java/mops/foren/infrastructure/persistence/mapper/UserMapper.java]: some naming

File: src/main/java/mops/foren/infrastructure/persistence/repositories/ForumRepositoryImpl.java [hunk 5]
```diff
@@ -44,4 +45,11 @@ public ForumRepositoryImpl(ForumJpaRepository forumRepository) {
                 .map(ForumMapper::mapForumDtoToForum)
                 .collect(Collectors.toList());
     }
+
+    @Override
+    public Forum getOneForumFromDB(ForumId forumId) {
+        return this.forumRepository.findById(forumId.getId()).stream()
+                .map(ForumMapper::mapForumDtoToForum)
+                .collect(Collectors.toList()).get(0);
+    }
```
Comment (niruc100) [inline src/main/java/mops/foren/infrastructure/persistence/repositories/ForumRepositoryImpl.java]: please refactor

File: src/main/java/mops/foren/infrastructure/persistence/repositories/ThreadJpaRepository.java [hunk 6]
```diff
@@ -0,0 +1,12 @@
...
+
+import mops.foren.infrastructure.persistence.dtos.ThreadDTO;
+import org.springframework.data.repository.CrudRepository;
+import org.springframework.stereotype.Repository;
+
+import java.util.List;
+
+@Repository
+public interface ThreadJpaRepository extends CrudRepository<ThreadDTO, Long> {
+    List<ThreadDTO> findByTopic_Id(Long id);
```
Comment (niruc100) [inline src/main/java/mops/foren/infrastructure/persistence/repositories/ThreadJpaRepository.java:11]: try renaming

File: src/main/java/mops/foren/infrastructure/persistence/repositories/ThreadRepositoryImpl.java [hunk 7]
```diff
@@ -0,0 +1,57 @@
...
+     * @param topic the topic the treads should be in.
+     * @return a list of threads with the wanted topic.
+     */
+    @Override
+    public List<Thread> getThreadsFromTopic(Topic topic) {
+        List<ThreadDTO> threadDtos = this.threadRepository.findByTopic_Id(topic.getId().getId());
+        return threadDtos.stream()
+                .map(ThreadMapper::mapThreadDtoToThread)
+                .collect(Collectors.toList());
+    }
```
Comment (niruc100) [inline src/main/java/mops/foren/infrastructure/persistence/repositories/ThreadRepositoryImpl.java]: please delete

**Review by niruc100 — CHANGES_REQUESTED** [review]
nice job

File: src/main/java/mops/foren/infrastructure/persistence/repositories/ThreadRepositoryImpl.java (hunk 7 above)
Comment (smncrm) [reply src/main/java/mops/foren/infrastructure/persistence/repositories/ThreadRepositoryImpl.java]: done

File: src/main/java/mops/foren/infrastructure/persistence/repositories/ForumRepositoryImpl.java (hunk 5 above)
Comment (smncrm) [reply src/main/java/mops/foren/infrastructure/persistence/repositories/ForumRepositoryImpl.java]: done

File: src/main/java/mops/foren/infrastructure/persistence/mapper/UserMapper.java (hunk 4 above)
Comment (smncrm) [reply src/main/java/mops/foren/infrastructure/persistence/mapper/UserMapper.java]: changed forenIds to forumIds

File: src/main/java/mops/foren/infrastructure/persistence/dtos/UserDTO.java (hunk 3 above)
Comment (smncrm) [reply src/main/java/mops/foren/infrastructure/persistence/dtos/UserDTO.java]: deleted cascadeType for now. To be decided in the future.

File: src/main/java/mops/foren/domain/repositoryabstraction/IThreadRepository.java (hunk 2 above)
Comment (smncrm) [reply src/main/java/mops/foren/domain/repositoryabstraction/IThreadRepository.java]: deleted

File: src/main/java/mops/foren/domain/repositoryabstraction/IPostRepository.java (hunk 1 above)
Comment (smncrm) [reply src/main/java/mops/foren/domain/repositoryabstraction/IPostRepository.java]: deleted

**Review by niruc100 — APPROVED** [review]

**Pull request merged**